### PR TITLE
Collective behavior trees

### PR DESCRIPTION
**Pull request by casals**
### Contains
- Additions and small changes in the `org.terasology.logic.behavior` package.

### Objective
To create a collective beehavior structure capable of interpreting a single behavior tree for a group of actors. This was necessary due to some restrictions found in the original group model while working on the last tasks of Milestone #2. The scenario is discussed in [the GSoC Milestone 2 blog post](https://casals.io/code/gsoc-reaching-second-milestone/).

### Changes

The following classes were added to the engine:

- `org.terasology.logic.behavior.core.CollectiveBehaviorTreeRunner`

- `org.terasology.logic.behavior.DefaultCollectiveBehaviorTreeRunner`

- `org.terasology.logic.behavior.CollectiveInterpreter`

- `org.terasology.logic.behavior.CollectiveBehaviorComponent`

Also: 

- the interface `Callback` was extracted from org.terasology.logic.behavior.DefaultBehaviorTreeRunner (not being used yet)

- copy constructors were added to the classes `CollectiveInterpreter` and `Interpreter`. This change is related to the task involving persistence of a behavior tree execution state for a single entity.

### Usage

- Install the module [WildAnimalsMadness] ([PR 31 for WildAnimals](https://github.com/Terasology/WildAnimals/pull/31) must be merged)

- Create/load a game using the module WildAnimalsMadness

- Execute the command `groupTestThree`

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by pollend** [comment]
can you remove all the .iml files those are generated by gradle when you run ./gradlew idea. those shouldn't be included in the PR.

**Comment by casals** [comment]
Hm, that should have been ignored - didn't realize it, sorry. Removing them now.

File: modules/CoreSampleGameplay/module.txt [hunk 1]
```diff
@@ -4,7 +4,8 @@
     "displayName" : "Core Gameplay",
     "description" : "Minimal gameplay template. Little content but a few starting items.",
     "dependencies" : [
-        {"id": "Core", "minVersion": "2.0.0"}
+        {"id": "Core", "minVersion": "2.0.0"},
+        {"id" : "WildAnimalsGenome", "minVersion" : "0.2.0"}
```
Comment (dkambersky) [inline modules/CoreSampleGameplay/module.txt]: Do we really want the CoreSampleGameplay module to pull in WildAnimalsGenome? At least I think it's something that should be discussed. Or is it left over from debugging?

Comment (casals) [reply modules/CoreSampleGameplay/module.txt]: Sorry, my mistake - leftovers from debugging, removed it.

**Comment by pollend** [comment]
this looks pretty good @dkambersky is it ok to merge this.

**Comment by dkambersky** [comment]
A 👍 from me

**Comment by Cervator** [comment]
> (but I don't have write access to merge it)

You do now! :-)

**Pull request merged**